lib/request_handler/body_parser.rb

```diff
@@ -3,14 +3,25 @@
 require 'request_handler/error'
 module RequestHandler
   class BodyParser < SchemaParser
-    def initialize(request:, schema:, schema_options: {})
+    def initialize(request:, schema:, schema_options: {}, included_schemas: {})
       raise MissingArgumentError, :"request.body" => 'is missing' if request.body.nil?
       super(schema: schema, schema_options: schema_options)
       @request = request
+      @included_schemas = included_schemas
     end
 
     def run
-      validate_schema(flattened_request_body)
+      flattened_body = flattened_request_body
```

Comment:
you could do here already
```ruby
body, *included = flattened_request_body
```

if `flattened_request_body` returns only a hash that will end up in `body` otherwise the multiple return values will be split up correctly